lib/ldclient-rb/impl/datasystem.rb

```diff
@@ -19,13 +19,12 @@ module DataSystem
       #
       # Starts the data system.
       #
-      # This method will return immediately. The provided event will be set when the system
+      # This method will return immediately. The returned event will be set when the system
       # has reached an initial state (either permanently failed, e.g. due to bad auth, or succeeded).
       #
-      # @param ready_event [Concurrent::Event] Event to set when initialization is complete
-      # @return [void]
+      # @return [Concurrent::Event] Event that will be set when initialization is complete
       #
-      def start(ready_event)
+      def start
```

Comment:
I was a little torn on having the ready_event passed in vs returning it. The existing data_source public interface for its start method returns a Concurrent::Event. This means we would either need an interface layer to sync the two events between the data system and the data source, or we would need a breaking change to have the data source interface accept the event vs return it.